packages/opentelemetry-plugin-mongodb-core/test/mongodb.test.ts

```diff
@@ -0,0 +1,273 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NodeTracer } from '@opentelemetry/node';
+import * as assert from 'assert';
+import * as mongodb from 'mongodb';
+import { plugin } from '../src';
+import { SpanKind } from '@opentelemetry/types';
+import { NoopLogger } from '@opentelemetry/core';
+import {
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+  ReadableSpan,
+} from '@opentelemetry/tracing';
+
+export interface MongoDBAccess {
+  client: mongodb.MongoClient;
+  collection: mongodb.Collection;
+}
+
+/**
+ * Access the mongodb collection.
+ * @param url The mongodb URL to access.
+ * @param dbName The mongodb database name.
+ * @param collectionName The mongodb collection name.
+ */
+function accessCollection(
+  url: string,
+  dbName: string,
+  collectionName: string
+): Promise<MongoDBAccess> {
+  return new Promise((resolve, reject) => {
+    mongodb.MongoClient.connect(url, function connectedClient(err, client) {
+      if (err) {
+        reject(err);
+        return;
+      }
+      const db = client.db(dbName);
+      const collection = db.collection(collectionName);
+      resolve({ client, collection });
+    });
+  });
+}
+
+/**
+ * Asserts root spans attributes.
+ * @param spans Readable spans that we need to asert.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpans(
+  spans: ReadableSpan[],
+  expectedName: string,
+  expectedKind: SpanKind
+) {
+  assert.strictEqual(spans.length, 2);
+  spans.forEach(span => {
+    assert(span.endTime instanceof Array);
+    assert(span.endTime.length === 2);
+  });
+  assert.strictEqual(spans[0].name, expectedName);
+  assert.strictEqual(spans[0].kind, expectedKind);
```

Comment:
Could you please check for `attributes` and `status`? Also, maybe use array destructuring instead of using index: `const [span1, span2] = spans`

Reply:
I've added a few asserts for attributes/status, however as @markwolff pointed out there is a specific API in `mongodb` module that should deprecate this instrumentation when implemented so i don't believe putting more work on tests is useful.